builtin/vault/config_sourcer.go

```diff
@@ -240,25 +240,31 @@ func (cs *ConfigSourcer) stop() error {
 	return nil
 }
 
-func (cs *ConfigSourcer) startRenewer(client *vaultapi.Client, path string, s *vaultapi.Secret) {
+func (cs *ConfigSourcer) startRenewer(client *vaultapi.Client, path string, s *vaultapi.Secret, log hclog.Logger) {
 	// The secret should be in the cache. If it isn't then just ignore.
 	// The reason it should be in the cache is because we only call startRenewer
 	// after querying the initial secret and inserting it into the cache.
+	log.Debug("checking if secret is in cache...")
 	cache, ok := cs.secretCache[path]
 	if !ok {
 		return
 	}
+	log.Debug("secret is in cache")
 
-	renewer, err := client.NewRenewer(&vaultapi.RenewerInput{
+	log.Debug("creating new lifetime watcher...")
+	renewer, err := client.NewLifetimeWatcher(&vaultapi.LifetimeWatcherInput{
```

Comment:
I think this is the main change right? It'd be nice if the commit message says what this changes and why it was broken before. I'm not entirely sure what this changes versus the `NewRenewer`

edit: Just peeped the Vault client source. Looks like `NewRenewer` was depreciated and called `NewLifetimeWatcher` anyways, so that seems like a good enough thing to include in the commit message

Reply:
Should this function also now be renamed `startLifetimeWatcher`?

Reply:
done and done!